### PR TITLE
Fix approval node documentation in `workflow_job_template` collection module

### PR DESCRIPTION
**Pull request by sean-m-sullivan**
##### SUMMARY
Fix approval node documentation in `workflow_job_template` collection module
Previously it had a reference that it was not implemented
Added example.

##### ISSUE TYPE
 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME
 - Collection

### CODE REVIEW

**Comment by AlanCoding** [comment]
I legitimately put this into a role and tried to run it. I get this.

```
Traceback (most recent call last):
  File "/home/alancoding/.ansible/tmp/ansible-tmp-1692973546.1473963-71605-92647199596413/AnsiballZ_workflow_job_template.py", line 107, in <module>
    _ansiballz_main()
  File "/home/alancoding/.ansible/tmp/ansible-tmp-1692973546.1473963-71605-92647199596413/AnsiballZ_workflow_job_template.py", line 99, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/home/alancoding/.ansible/tmp/ansible-tmp-1692973546.1473963-71605-92647199596413/AnsiballZ_workflow_job_template.py", line 47, in invoke_module
    runpy.run_module(mod_name='ansible_collections.awx.awx.plugins.modules.workflow_job_template', init_globals=dict(_module_fqn='ansible_collections.awx.awx.plugins.modules.workflow_job_template', _modlib_path=modlib_path),
  File "<frozen runpy>", line 226, in run_module
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/tmp/ansible_awx.awx.workflow_job_template_payload_8k3ok9mm/ansible_awx.awx.workflow_job_template_payload.zip/ansible_collections/awx/awx/plugins/modules/workflow_job_template.py", line 1008, in <module>
  File "/tmp/ansible_awx.awx.workflow_job_template_payload_8k3ok9mm/ansible_awx.awx.workflow_job_template_payload.zip/ansible_collections/awx/awx/plugins/modules/workflow_job_template.py", line 1001, in main
  File "/tmp/ansible_awx.awx.workflow_job_template_payload_8k3ok9mm/ansible_awx.awx.workflow_job_template_payload.zip/ansible_collections/awx/awx/plugins/modules/workflow_job_template.py", line 779, in create_workflow_nodes_association
TypeError: string indices must be integers, not 'str'
```

Additionally, error handling makes things hard, and I found this was needed

```diff
diff --git a/awx_collection/plugins/modules/workflow_job_template.py b/awx_collection/plugins/modules/workflow_job_template.py
index c8ee6e7995..d77faba3de 100644
--- a/awx_collection/plugins/modules/workflow_job_template.py
+++ b/awx_collection/plugins/modules/workflow_job_template.py
@@ -632,7 +632,7 @@ def create_workflow_nodes(module, response, workflow_nodes, workflow_id):
                 workflow_node_fields['unified_job_template'] = unified_job_template['id']
             else:
                 if workflow_node['unified_job_template']['type'] != 'workflow_approval':
-                    module.fail_json(msg="Unable to Find unified_job_template: {0}".format(search_fields))
+                    module.fail_json(msg="Unable to Find unified_job_template '{0}': {1}".format(workflow_node['unified_job_template']['name'], search_fields))
 
         # Lookup Values for other fields
 
```

**Comment by AlanCoding** [comment]
This example seems borderline beyond repair

```diff
diff --git a/awx_collection/plugins/modules/workflow_job_template.py b/awx_collection/plugins/modules/workflow_job_template.py
index c8ee6e7995..fbd21f9742 100644
--- a/awx_collection/plugins/modules/workflow_job_template.py
+++ b/awx_collection/plugins/modules/workflow_job_template.py
@@ -632,7 +632,7 @@ def create_workflow_nodes(module, response, workflow_nodes, workflow_id):
                 workflow_node_fields['unified_job_template'] = unified_job_template['id']
             else:
                 if workflow_node['unified_job_template']['type'] != 'workflow_approval':
-                    module.fail_json(msg="Unable to Find unified_job_template: {0}".format(search_fields))
+                    module.fail_json(msg="Unable to Find unified_job_template '{0}': {1}".format(workflow_node['unified_job_template']['name'], search_fields))
 
         # Lookup Values for other fields
```

This was needed in order to assess the next error - which is that `example-project` must be an inventory source (like, what?) and inventory sources cannot accept prompts.

I could fix the prior error by specifying credentials like:

```yaml
          credentials:
            - name: local_cred
            - name: suplementary cred
```

As opposed to what we had.

I don't see why we're keeping around examples that aren't being ran. I'd say delete the whole thing and reference the tests instead. The tests, for their part, only use `workflow_nodes:` once. I don't know how we manage to add these features without testing.

**Comment by sean-m-sullivan** [comment]
I took your advice, and updated the test to include all the examples we needed, and then substituted out the variables for example names for the example.

Hopefully this makes it easier for everyone.

File: awx_collection/plugins/modules/workflow_job_template.py [hunk 1]
```diff
@@ -517,68 +517,63 @@
...
                 organization:
                     name: Default
-  register: result
+      - all_parents_must_converge: false
+        identifier: node301
+        unified_job_template:
+            description: Approval node for example
+            timeout: 900
+            type: workflow_approval
+            name: "{{ approval_node_name }}"
```
Comment (AlanCoding) [inline awx_collection/plugins/modules/workflow_job_template.py]: Did you mean to leave this templated? (going by what you said in your comment)

I'm not highly adverse to this, just wanted to call it out.

**Review by AlanCoding — APPROVED** [review]
Confirmed locally that the integration test is passing, which does a lot to give confidence in the example.

File: awx_collection/plugins/modules/workflow_job_template.py (hunk 1 above)
Comment (sean-m-sullivan) [reply awx_collection/plugins/modules/workflow_job_template.py]: I did not going to fix.

Comment (sean-m-sullivan) [reply awx_collection/plugins/modules/workflow_job_template.py]: Fixed

**Pull request merged**